WordPress/src/main/java/org/wordpress/android/ui/suggestion/service/SuggestionService.java

```diff
@@ -47,27 +50,43 @@ public IBinder onBind(Intent intent) {
+    public void update(final long siteId) {
+        boolean currentlyRequestingSuggestions = mCurrentlyRequestingSuggestionsSiteIds.contains(siteId);
+        boolean suggestionsAreStale = SUGGESTION_REQUEST_THROTTLER.areResultsStale(siteId);
+        if (!currentlyRequestingSuggestions && suggestionsAreStale) {
+            updateSuggestions(siteId);
+        } else {
+            String reason = currentlyRequestingSuggestions
+                    ? "a suggestions request is already in progress."
+                    : "the suggestions were recently updated.";
+            AppLog.d(T.SUGGESTION, "Skipping suggestion update for site " + siteId + " because " + reason);
+        }
+
+        boolean currentlyRequestingTags = mCurrentlyRequestingTagsSiteIds.contains(siteId);
+        boolean tagsAreStale = TAG_REQUEST_THROTTLER.areResultsStale(siteId);
+        if (!currentlyRequestingTags && tagsAreStale) {
+            updateTags(siteId);
+        } else {
+            String reason = currentlyRequestingTags
+                    ? "a tags request is already in progress."
+                    : "the tags were recently updated.";
+            AppLog.d(T.SUGGESTION, "Skipping tags update for site " + siteId + " because " + reason);
+        }
+    }
+
+    // We're still using `siteId` here and not a SiteModel, because suggestions the endpoint works on any site, not only
     // on sites the user is a member of. Username suggestions are used in the Reader to auto complete usernames
     // when replying to comments.
-    public void updateSuggestions(final long siteId) {
-        if (mCurrentlyRequestingSuggestionsSiteIds.contains(siteId)) {
-            return;
-        }
+    private void updateSuggestions(final long siteId) {
         mCurrentlyRequestingSuggestionsSiteIds.add(siteId);
-        RestRequest.Listener listener = new RestRequest.Listener() {
-            @Override
-            public void onResponse(JSONObject jsonObject) {
-                handleSuggestionsUpdatedResponse(siteId, jsonObject);
-                removeSiteIdFromSuggestionRequestsAndStopServiceIfNecessary(siteId);
-            }
+        RestRequest.Listener listener = jsonObject -> {
+            handleSuggestionsUpdatedResponse(siteId, jsonObject);
+            SUGGESTION_REQUEST_THROTTLER.onResponseReceived(siteId);
+            removeSiteIdFromSuggestionRequestsAndStopServiceIfNecessary(siteId);
         };
-        RestRequest.ErrorListener errorListener = new RestRequest.ErrorListener() {
-            @Override
-            public void onErrorResponse(VolleyError volleyError) {
-                AppLog.e(AppLog.T.SUGGESTION, volleyError);
-                removeSiteIdFromSuggestionRequestsAndStopServiceIfNecessary(siteId);
-            }
+        RestRequest.ErrorListener errorListener = volleyError -> {
+            AppLog.e(AppLog.T.SUGGESTION, volleyError);
+            removeSiteIdFromSuggestionRequestsAndStopServiceIfNecessary(siteId);
```

Comment:
There is no change to the logic in these listeners, this is just switching to using the lambda syntax.